src/components/chips/chips.spec.js

```diff
@@ -213,20 +213,40 @@ describe('<md-chips>', function() {
         expect(scope.removeChip.calls.mostRecent().args[1]).toBe(0);       // Index
       });
 
+      var CHIP_COMPONENT_TEMPLATE =
+        '<md-chips ng-model="items" md-on-chip-select="onChipSelect($index)" md-on-chip-deselect="onChipDeselect($index)"></md-chips>';
 
-      it('should call the select method when selecting a chip', function() {
-        var element = buildChips(CHIP_SELECT_TEMPLATE);
+      it("should call the select and deselect expressions", function(done) {
+        var element = buildChips(CHIP_COMPONENT_TEMPLATE);
         var ctrl = element.controller('mdChips');
+        var scope = element.scope();
+        var state = 0;
```

Comment:
What exactly is the `state` variable here? It looks like it controls the expectations below, but you should never have a conditional expectation within a test. If you need to have different expectations, these should be in different tests.